projects/core/src/cart/store/effects/cart-entry.effect.ts

```diff
@@ -115,8 +173,61 @@ export class CartEntryEffects {
     )
   );
 
+  @Effect()
+  updateEntryByKey$: Observable<
+    | CartActions.CartUpdateEntryByKeySuccess
+    | CartActions.CartUpdateEntryByKeyFail
+    | DeprecatedCartActions.LoadCart
+  > = this.actions$.pipe(
+    ofType(CartActions.CART_UPDATE_ENTRY_BY_KEY),
+    map((action: CartActions.CartUpdateEntryByKey) => action.payload),
+    concatMap(payload =>
+      // TODO add take until for contextChange
+      this.cartConnector.load(payload.userId, payload.cartId).pipe(
+        switchMap(cart => {
+          const entryToUpdate = cart.entries.find(
+            entry => entry.key === payload.key
+          );
+          return this.cartEntryConnector
+            .update(
+              payload.userId,
+              payload.cartId,
+              `${entryToUpdate.entryNumber}`,
+              payload.quantity
+            )
+            .pipe(
+              mergeMap(() => {
```

Comment:
Is `mergeMap` necessary? Can we simplify it to `mapTo(from([ <actions here> ])`?